### PR TITLE
fix:  prevent creation of discussions with whitespace-only names

### LINKED ISSUE #38434: Discussion can be created with whitespace-only name
### Description

When creating a discussion, the Name field accepts whitespace-only input (spaces/tabs).
Submitting the form with such input successfully creates a discussion with an effectively empty name.
This results in unnamed discussions appearing in the sidebar, which can be confusing and hard to manage.
see 👇

Image

### Steps to reproduce
1. Click + → Create discussion
2. Select a Parent channel or team
3. In the Name field, enter only spaces (e.g. " ")
4. Click Create

### Expected behavior

1.  The discussion should not be created if the Name contains only whitespace.
2. The Name field should be validated after trimming whitespace and show an error (similar to an empty value).

### Actual behavior

1.The discussion is created successfully.
2. The discussion appears in the sidebar without a visible name.

### Why this matters
1. The Name field is required, but whitespace-only input bypasses validation.
2. This can lead to:
                      i) Unidentifiable discussion rooms
                      ii) Confusing navigation for users and admins
                      iii) Potential moderation and accessibility issues
 
### Environment
1. Rocket.Chat Web
2. Tested locally
3. Browser: Chrome
4. OS: Any

### Evidence

https://github.com/user-attachments/assets/c7696aad-7474-44bb-9397-0443496998e4

### PR DESCRIPTION
**Pull request by anupam-pandey01**
## Summary: 
Added validation to the CreateDiscussion component to prevent the creation of discussions with names consisting only of whitespace (spaces/tabs). This ensures that all discussions in the sidebar have a visible, identifiable name.

## Proposed changes (including videos or screenshots):

- Updated the create button state/validation logic to treat whitespace-only strings as empty.
- Applied .trim() to the name input before validation.
- Added a changeset file to ensure this fix is included in the next release notes.

https://github.com/user-attachments/assets/3a8729c7-e57f-40e2-bfb0-3d765872322f

## Issue(s):
Closes #38434

## Steps to test or reproduce:

1. Go to any channel and click the + icon to Create Discussion.
2. Leave the "Name" field empty or enter only spaces (e.g., " ").
3. Enter Create

## Summary by CodeRabbit

* **Bug Fixes**
  * Prevent creation of discussions whose names are empty or consist only of whitespace by trimming input before validation.
  * Show a clear, explicit error when a submitted name is invalid.
  * Improve form validation and input accessibility so users receive immediate, accessible guidance before submission.

### CODE REVIEW

**Comment by ggazzo** [comment]
What do you think about also validating on the backend?

**Comment by dougfabris** [comment]
@anupam-pandey01 can we add a validation in the backend as well?

**Comment by anupam-pandey01** [comment]
> @anupam-pandey01 can we add a validation in the backend as well?

Thanks for the suggestion.
I’m working on adding backend validation and will update the PR shortly.

**Comment by anupam-pandey01** [comment]
@dougfabris Added frontend and backend validation to prevent whitespace.
Please let me know if any changes are needed.

File: apps/meteor/app/discussion/server/methods/createDiscussion.ts [hunk 1]
```diff
@@ -74,6 +74,19 @@ const create = async ({
 	encrypted,
 	topic,
 }: CreateDiscussionProperties): Promise<IRoom & { rid: string }> => {
+	// Validate discussion name to reject empty or whitespace-only values.
```
Comment (KevLehman) [inline apps/meteor/app/discussion/server/methods/createDiscussion.ts]: pls add api tests

File: apps/meteor/app/discussion/server/methods/createDiscussion.ts [hunk 2]
```diff
@@ -74,6 +74,19 @@ const create = async ({
...
+	if (!discussionName || discussionName.trim().length === 0) {
+		throw new Meteor.Error(
+			'error-invalid-discussion-name',
+			'Discussion name cannot be empty', { 
+				method: 'DiscussionCreation'
```
Comment (KevLehman) [inline apps/meteor/app/discussion/server/methods/createDiscussion.ts]:
```suggestion
				method: 'createDiscussion' 
```

File: apps/meteor/app/discussion/server/methods/createDiscussion.ts [hunk 3]
```diff
@@ -74,6 +74,19 @@ const create = async ({
...
+		throw new Meteor.Error(
+			'error-invalid-discussion-name',
+			'Discussion name cannot be empty', { 
+				method: 'DiscussionCreation' 
+			},
+		);
+	}
+
+	// Remove leading and trailing whitespace
+	discussionName = discussionName.trim();
```
Comment (KevLehman) [inline apps/meteor/app/discussion/server/methods/createDiscussion.ts]: Do this before the if so you don't have to trim twice

Comment (KevLehman) [reply apps/meteor/app/discussion/server/methods/createDiscussion.ts]: discusionName = discussionName?.trim() (or smth)

**Review by KevLehman — CHANGES_REQUESTED** [review]

File: apps/meteor/app/discussion/server/methods/createDiscussion.ts (hunk 1 above)
Comment (anupam-pandey01) [reply apps/meteor/app/discussion/server/methods/createDiscussion.ts]: @KevLehman could you please guide me on where the test file should be located or which existing test file I should extend ?

**Comment by anupam-pandey01** [comment]
i have done all changes. could anyone please guide me on where the test file should be located or which existing test file I should extend ?

File: apps/meteor/app/discussion/server/methods/createDiscussion.ts (hunk 1 above)
Comment (KevLehman) [reply apps/meteor/app/discussion/server/methods/createDiscussion.ts]: Try searching for existing tests on folders named `test`, `tests`, etc. That way you get to learn the codebase a bit better.

Comment (anupam-pandey01) [reply apps/meteor/app/discussion/server/methods/createDiscussion.ts]: There is one test folder in codebase. i think this is right place  'Rocket.Chat\apps\meteor\tests\unit\app'

**Review by anupam-pandey01 — COMMENTED** [review]
@KevLehman I have added frontend and backend validation add also added api test for whitespace-only discussion names.
Please review 🙏
